crates/bevy_reflect/bevy_reflect_derive/src/derive_data.rs

```diff
@@ -444,6 +455,7 @@ impl<'a> ReflectStruct<'a> {
     }
 
     /// Get a collection of types which are ignored by the reflection API
+    #[allow(dead_code)]
```

Comment:
A note here and below about why we've left in dead code here would be nice.

Or you can remove it!

Reply:
Removed!

Note: I only removed the ones that I added `#[dead_code]` to. There are still others in this file, but I won't remove them here since they're otherwise unaffected by this PR.